src/scanpy/plotting/_baseplot_class.py

```diff
@@ -74,141 +81,126 @@ class BasePlot:
-
+    DEFAULT_SAVE_PREFIX: ClassVar[str] = "baseplot_"
+    # maximum number of categories allowed to be plotted
+    MAX_NUM_CATEGORIES: ClassVar[int] = 500
+
+    adata: AnnData
+    var_names: _VarNames | Mapping[str, _VarNames]
+    groupby: str | Sequence[str]
+    _: KW_ONLY
+    use_raw: bool | None = None
+    log: bool = True
+    num_categories: int = 5
+    categories_order: Sequence[str] | None = None
+    title: str | None = None
+    figsize: tuple[float, float] | None = None
+    gene_symbols: str | None = None
+    var_group_positions: Sequence[tuple[int, int]] | None = None
+    var_group_labels: Sequence[str] | None = None
+    var_group_rotation: float | None = None
+    layer: str | None = None
+    ax: Axes | None = None
+    vmin: float | None = None
+    vmax: float | None = None
+    vcenter: float | None = None
+    norm: Normalize | None = None
+
+    # minimum height required for legends to plot properly
+    min_figure_height: float = 2.5
+    category_height = 0.35
+    category_width = 0.37
     # gridspec parameter. Sets the space between mainplot, dendrogram and legend
-    DEFAULT_WSPACE = 0
-
-    DEFAULT_COLORMAP = "winter"
-    DEFAULT_LEGENDS_WIDTH = 1.5
-    DEFAULT_COLOR_LEGEND_TITLE = "Expression\nlevel in group"
-
-    MAX_NUM_CATEGORIES = 500  # maximum number of categories allowed to be plotted
-
-    @old_positionals(
-        "use_raw",
-        "log",
-        "num_categories",
-        "categories_order",
-        "title",
-        "figsize",
-        "gene_symbols",
-        "var_group_positions",
-        "var_group_labels",
-        "var_group_rotation",
-        "layer",
-        "ax",
-        "vmin",
-        "vmax",
-        "vcenter",
-        "norm",
+    wspace: float = 0
+    cmap: str | None = "winter"
+    legends_width: float = 1.5
+    color_legend_title: str = "Expression\nlevel in group"
+    are_axes_swapped: bool = False
+    var_names_idx_order: Sequence[int] | None = None
+    group_extra_size: float = 0.0
+    plot_group_extra: dict[str, object] | None = None
+    # after .render() is called the fig value is assigned and ax_dict
+    # contains a dictionary of the axes used in the plot
+    fig: Figure | None = None
+    ax_dict: dict[str, Axes] | None = None
+
+    kwds: Mapping[str, object] = field(default_factory=dict)
+
+    # properties aliasing fields
+    @property
+    def has_var_groups(self) -> bool:
+        return len(self.var_group_positions or ()) > 0
+
+    @property
+    def fig_title(self) -> str | None:
+        return self.title
+
+    @property
+    def width(self) -> float | None:
+        return self.figsize[0] if self.figsize is not None else None
+
+    @property
+    def height(self) -> float | None:
+        return self.figsize[1] if self.figsize is not None else None
+
+    @property
+    def vboundnorm(self) -> VBoundNorm:
+        return VBoundNorm(self.vmin, self.vmax, self.vcenter, self.norm)
+
+    # deprecated class vars
```

Comment:
Why deprecated? Because they are already set from elsewhere (the function-based API)?

Reply:
If so, why re-introduce?

Reply:
So these are part of the public API, the idea being to get rid of these eventually? https://scanpy.readthedocs.io/en/stable/api/generated/classes/scanpy.pl.StackedViolin.html

Reply:
Yeah, the default would be accessible like `default = Baseplot.cmap`, and overridable in subclasses:

```py
class DotPlot(BasePlot):
   cmap = "Blues"
```

or just by setting it on the instance.